core/inc/knoting/px_variables_wrapper.h

```diff
@@ -0,0 +1,185 @@
+#pragma once
+#include <PxPhysicsAPI.h>
+
+using namespace physx;
+
+namespace knot {
+class PxScene_ptr_wrapper {
+   public:
+    PxScene_ptr_wrapper(PxScene* scene = nullptr) : m_px_scene(scene) {}
+    ~PxScene_ptr_wrapper() {
+        if (m_px_scene) {
+            m_px_scene->release();
+            m_px_scene = nullptr;
+        }
+    }
+
+    void set(PxScene* scene) { m_px_scene = scene; }
+
+    PxScene* get() { return m_px_scene; }
+
+   private:
+    PxScene* m_px_scene;
+};
+
+class PxPhysics_ptr_wrapper {
+   public:
+    PxPhysics_ptr_wrapper(PxPhysics* physics = nullptr) : m_px_physics(physics) {}
+    ~PxPhysics_ptr_wrapper() {
+        if (m_px_physics) {
+            m_px_physics->release();
+            m_px_physics = nullptr;
+        }
+    }
+
+    void set(PxPhysics* physics) {
+        if (m_px_physics) {
+            m_px_physics->release();
+            m_px_physics = nullptr;
+        }
```

Comment:
_I_ would the release and nullptr inside a `clean` function, to avoid having this code repeated twice. But it is not 100% necessary.